table/tables/tables.go

```diff
@@ -129,44 +127,39 @@ func TableFromMeta(alloc autoid.Allocator, tblInfo *model.TableInfo) (table.Tabl
 	}
 
 	var t Table
-	initTableCommon(&t.tableCommon, tblInfo.ID, tblInfo.ID, columns, alloc)
-
-	for _, idxInfo := range tblInfo.Indices {
-		if idxInfo.State == model.StateNone {
-			return nil, table.ErrIndexStateCantNone.Gen("index %s can't be in none state", idxInfo.Name)
-		}
-
-		idx := NewIndex(tblInfo, idxInfo)
-		t.indices = append(t.indices, idx)
+	if err := initTableCommon(&t.tableCommon, tblInfo, tblInfo.ID, columns, alloc); err != nil {
```

Comment:
In the case of partitions, it seems to have built two indexes here. The first time is here, the other time is when building the index(https://github.com/pingcap/tidb/pull/7062/files#diff-c709fb799ac0eac6d7ea40b2278c90d7R69).